edk2toollib/utility_functions_test.py

```diff
@@ -65,5 +65,32 @@ def test_locate_class_in_module(self):
         self.assertEqual(found_class, GrandChildOfDesiredClass)
 
 
+class EnumClassTest(unittest.TestCase):
+    def test_EnumClass_array(self):
+        all_animals = ["Dog", "Cat", "Rabbit"]
+        animals = utilities.Enum(all_animals)
+        
+        # make sure we have three animals
+        self.assertEqual(len(animals), 3)
+        self.assertTrue(hasattr(animals, "Dog"))
+        self.assertFalse(hasattr(animals, "Rat"))
+        self.assertFalse(hasattr(animals, "dog"))
+        # check to make sure the values are unique
+        self.assertNotEqual(animals.Dog, animals.Cat)
+
+        # make sure we can iterate over the members
+        for animal in animals.__members__:
+            self.assertIn(animal, all_animals)
+
+
+    def test_EnumClass_args(self):
```

Comment:
Can we add a test for "not in enum"?

Reply:
We do have a test to make sure it's not in.
``` python
self.assertFalse(hasattr(animals, "Rat"))
self.assertFalse(hasattr(animals, "dog"))
```

We could create two seperate enums and check to make sure they aren't in each other

Reply:
Didn't notice that.